File: 1-feed-caching-with-redis/2-csharp/CachedPost.cs
using Microsoft.EntityFrameworkCore;

namespace FeedCacheService;

/// <summary>Post entity — source rows before ZADD into Redis ZSET.</summary>
public class CachedPost
{
    public string Id { get; set; } = default!;
    public string AuthorId { get; set; } = default!;
    public string Content { get; set; } = default!;

    // Epoch milliseconds used as ZSET score; stored as bigint.
    public long ScoreMs { get; set; }
}

/// <summary>EF Core context mapping posts table.</summary>
public class FeedDbContext : DbContext
{
    public FeedDbContext(DbContextOptions<FeedDbContext> options) : base(options)
    {
    }

    public DbSet<CachedPost> Posts => Set<CachedPost>();

    protected override void OnModelCreating(ModelBuilder modelBuilder)
    {
        var post = modelBuilder.Entity<CachedPost>();
        post.ToTable("posts");
        post.HasKey(p => p.Id);
        post.Property(p => p.Id).HasColumnName("id");
        post.Property(p => p.AuthorId).HasColumnName("author_id");
        post.Property(p => p.Content).HasColumnName("content");
        post.Property(p => p.ScoreMs).HasColumnName("score_ms");
    }
}

File: 1-feed-caching-with-redis/1-java/src/main/java/com/starci/feedcache/CachedPost.java
package com.starci.feedcache;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

/**
 * Post entity — source rows before ZADD into Redis ZSET.
 */
@Entity
@Table(name = "posts")
public class CachedPost {
    @Id
    private String id;

    @Column(name = "author_id")
    private String authorId;

    @Column
    private String content;

    // Epoch milliseconds used as ZSET score; stored as bigint.
    @Column(name = "score_ms")
    private long scoreMs;

    public CachedPost() {
    }

    public CachedPost(String id, String authorId, String content, long scoreMs) {
        this.id = id;
        this.authorId = authorId;
        this.content = content;
        this.scoreMs = scoreMs;
    }

    public String getId() {
        return id;
    }

    public String getAuthorId() {
        return authorId;
    }

    public String getContent() {
        return content;
    }

    public long getScoreMs() {
        return scoreMs;
    }
}
